exporter/otlp-metrics/lib/opentelemetry/exporter/otlp/metrics/metrics_exporter.rb

```diff
@@ -0,0 +1,307 @@
+                  aggregation_temporality: as_otlp_aggregation_temporality(metrics.aggregation_temporality),
+                  data_points: metrics.data_points.map do |ndp|
+                    number_data_point(ndp)
+                  end
+                )
+              )
+
+            when :counter, :up_down_counter
+              Opentelemetry::Proto::Metrics::V1::Metric.new(
+                name: metrics.name,
+                description: metrics.description,
+                unit: metrics.unit,
+                sum: Opentelemetry::Proto::Metrics::V1::Sum.new(
+                  aggregation_temporality: as_otlp_aggregation_temporality(metrics.aggregation_temporality),
+                  data_points: metrics.data_points.map do |ndp|
+                    number_data_point(ndp)
+                  end
+                )
+              )
+
+            when :histogram
+              Opentelemetry::Proto::Metrics::V1::Metric.new(
+                name: metrics.name,
+                description: metrics.description,
+                unit: metrics.unit,
+                histogram: Opentelemetry::Proto::Metrics::V1::Histogram.new(
+                  aggregation_temporality: as_otlp_aggregation_temporality(metrics.aggregation_temporality),
+                  data_points: metrics.data_points.map do |hdp|
+                    histogram_data_point(hdp)
+                  end
+                )
+              )
+            end
+          end
+
+          def as_otlp_aggregation_temporality(type)
+            case type
+            when :delta then Opentelemetry::Proto::Metrics::V1::AggregationTemporality::AGGREGATION_TEMPORALITY_DELTA
+            when :cumulative then Opentelemetry::Proto::Metrics::V1::AggregationTemporality::AGGREGATION_TEMPORALITY_CUMULATIVE
+            else Opentelemetry::Proto::Metrics::V1::AggregationTemporality::AGGREGATION_TEMPORALITY_UNSPECIFIED
+            end
+          end
+
+          def histogram_data_point(hdp)
+            Opentelemetry::Proto::Metrics::V1::HistogramDataPoint.new(
+              attributes: hdp.attributes.map { |k, v| as_otlp_key_value(k, v) },
+              start_time_unix_nano: hdp.start_time_unix_nano,
+              time_unix_nano: hdp.time_unix_nano,
+              count: hdp.count,
+              sum: hdp.sum,
+              bucket_counts: hdp.bucket_counts,
+              explicit_bounds: hdp.explicit_bounds,
+              exemplars: hdp.exemplars,
+              min: hdp.min,
+              max: hdp.max
+            )
+          end
+
+          def number_data_point(ndp)
+            Opentelemetry::Proto::Metrics::V1::NumberDataPoint.new(
+              attributes: ndp.attributes.map { |k, v| as_otlp_key_value(k, v) },
+              as_int: ndp.value,
+              start_time_unix_nano: ndp.start_time_unix_nano,
+              time_unix_nano: ndp.time_unix_nano,
+              exemplars: ndp.exemplars # exemplars not implemented yet from metrics sdk
+            )
+          end
+
```

Comment:
For spec compliance, we need a `force_flush` method. I think it's safe to mirror what's done in the OTLP traces exporter.

https://github.com/open-telemetry/opentelemetry-ruby/blob/406cabe54f8fb4937d0f6b73fe830a51c04fdb92/exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb#L83-L90

Reply:
Added